### PR TITLE
Merge/update Swift 3 branch

### PR DESCRIPTION
**Pull request by ewerx**
Needed this branch to test something so I merged master and updated the pods to get it working.

### CODE REVIEW

**Comment by jpsim** [comment]
Thanks for the PR, @ewerx! @TimOliver's responsible for this repo, but he's on holiday break right now, so hopefully it won't be too long before he's able to review.

File: RealmTasks Apple/Podfile [hunk 1]
```diff
@@ -2,27 +2,20 @@ source 'https://github.com/CocoaPods/Specs.git'
...
+
     target 'RealmTasks iOS Tests' do
         platform :ios, '9.0'
     end
 end
-
-post_install do |installer|
-  installer.pods_project.targets.each do |target|
-    target.build_configurations.each do |config|
-      config.build_settings['SWIFT_VERSION'] = '3.0'
```
Comment (jpsim) [inline RealmTasks Apple/Podfile]: without this, I get a build error for the Cartography pod:

> “Use Legacy Swift Language Version” (SWIFT_VERSION) is required to be configured correctly for targets which use Swift. Use the [Edit > Convert > To Current Swift Syntax…] menu to choose a Swift version or use the Build Settings editor to configure the build setting directly.

**Comment by jpsim** [comment]
It also seems that `SWIFT_VERSION` needs to be changed from 3.0.1 to 3.0:

```diff
diff --git a/RealmTasks Apple/RealmTasks.xcodeproj/project.pbxproj b/RealmTasks Apple/RealmTasks.xcodeproj/project.pbxproj
index 8b4cde9..d75a5fe 100644
--- a/RealmTasks Apple/RealmTasks.xcodeproj/project.pbxproj	
+++ b/RealmTasks Apple/RealmTasks.xcodeproj/project.pbxproj	
@@ -737,7 +737,7 @@
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
 				PRODUCT_BUNDLE_IDENTIFIER = "io.realm.RealmTasks-iOS-Tests";
 				PRODUCT_NAME = "$(TARGET_NAME)";
-				SWIFT_VERSION = 3.0.1;
+				SWIFT_VERSION = 3.0;
 				TEST_HOST = "$(BUILT_PRODUCTS_DIR)/RealmTasks.app/RealmTasks";
 			};
 			name = Debug;
@@ -753,7 +753,7 @@
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
 				PRODUCT_BUNDLE_IDENTIFIER = "io.realm.RealmTasks-iOS-Tests";
 				PRODUCT_NAME = "$(TARGET_NAME)";
-				SWIFT_VERSION = 3.0.1;
+				SWIFT_VERSION = 3.0;
 				TEST_HOST = "$(BUILT_PRODUCTS_DIR)/RealmTasks.app/RealmTasks";
 			};
 			name = Release;
@@ -771,7 +771,7 @@
 				MACOSX_DEPLOYMENT_TARGET = 10.10;
 				PRODUCT_BUNDLE_IDENTIFIER = io.realm.realmtasks.macos;
 				SDKROOT = macosx;
-				SWIFT_VERSION = 3.0.1;
+				SWIFT_VERSION = 3.0;
 			};
 			name = Debug;
 		};
@@ -788,7 +788,7 @@
 				MACOSX_DEPLOYMENT_TARGET = 10.10;
 				PRODUCT_BUNDLE_IDENTIFIER = io.realm.realmtasks.macos;
 				SDKROOT = macosx;
-				SWIFT_VERSION = 3.0.1;
+				SWIFT_VERSION = 3.0;
 			};
 			name = Release;
 		};
@@ -835,6 +835,7 @@
 				OTHER_SWIFT_FLAGS = "-DDEBUG";
 				PRODUCT_NAME = RealmTasks;
 				SWIFT_OPTIMIZATION_LEVEL = "-Onone";
+				SWIFT_VERSION = 3.0;
 			};
 			name = Debug;
 		};
@@ -873,6 +874,7 @@
 				MTL_ENABLE_DEBUG_INFO = NO;
 				PRODUCT_NAME = RealmTasks;
 				SWIFT_OPTIMIZATION_LEVEL = "-Owholemodule";
+				SWIFT_VERSION = 3.0;
 				VALIDATE_PRODUCT = YES;
 			};
 			name = Release;
@@ -890,7 +892,7 @@
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
 				PRODUCT_BUNDLE_IDENTIFIER = io.realm.realmtasks.ios;
 				SDKROOT = iphoneos;
-				SWIFT_VERSION = 3.0.1;
+				SWIFT_VERSION = 3.0;
 				TARGETED_DEVICE_FAMILY = "1,2";
 			};
 			name = Debug;
@@ -908,7 +910,7 @@
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
 				PRODUCT_BUNDLE_IDENTIFIER = io.realm.realmtasks.ios;
 				SDKROOT = iphoneos;
-				SWIFT_VERSION = 3.0.1;
+				SWIFT_VERSION = 3.0;
 				TARGETED_DEVICE_FAMILY = "1,2";
 			};
 			name = Release;
```

**Comment by TimOliver** [comment]
Thanks for the PR! I just finished updating #357 to the latest version on `master` and addressed all of the feedback in that one. I think at this point, that one should be slightly further ahead than this one now.

Considering this PR was mainly just doing the same in terms of updating pods and the latest branch, shall I just copy and paste over any other improvements to #357?

**Comment by ewerx** [comment]
I didn't make any other improvements, just the merge and update. Feel free to close this PR.

**Pull request closed without merging**